addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2438,6 +2438,15 @@ options.registry.CookiesBar = options.registry.SnippetPopup.extend({
         ['/website/static/src/xml/website.cookies_bar.xml']
     ),
 
+    /**
+     * @override
+     */
+    async willStart() {
+        // We hide the cookie bar on start because it prevents from snippet dropping in the main page
+        await this._super(...arguments);
+        return this.onTargetHide();
+    },
```

Comment:
Strange, how does this actually work? In my mind the options should not be instantiated until we click on the related snippet so this should not be called on editor start... maybe invisible snippet are pre-instantiated so that their onTargetShow/Hide methods can be called?

Reply:
This might be the case because of this? https://github.com/odoo/odoo/blob/7aa08bfb6b2e316b311cc5be81a63eacb9fa2556/addons/web_editor/static/src/js/editor/snippets.editor.js#L1663 

Took me a while to get it because I mostly worked with o_snippet_invisible so I thought options were initialised as soon as we show the SnippetMenu. 😅

Since it only works because of the way updateInvisibleDOM init an editor, it might be risky to implement it this way as it's not really expected behaviour.

Not sure how we could hide it on loading then.

Reply:
Yes that is what I imagined :wink: 

> Not sure how we could hide it on loading then.

The same code which shows the cookie bar in non edit mode should hide it when entering edit mode (the cookie/popup "public widget", destroy is called when entering edit mode).

Reply:
Hiding the modal also triggers the consent event. I don't think not getting the consent of the designer is a big problem but it would also mean that if they trigger the cookie bar and go directly into the editor, they would almost never see the cookie bar if they don't manually enable it.

Hiding it manually also triggers the consent event but at least we're sure that they've seen it once. We could maybe have a tour that tells them to edit the cookie bar if they want to and then click on the hide button to continue editing the rest of the website maybe?

Reply:
Anyways, that would have been overthinking it :wink: Cookie bar enabled/disabled as an editor user is not important and we would not want a tour to explain that to the user anyway (I suppose you saw the modal which allows to enable the cookie bar... FP did not really want to encourage to use it ;))